File: datenow/src/main/java/com/grepp/datenow/infra/error/CommonException.java
package com.grepp.datenow.infra.error;

import com.grepp.datenow.infra.response.ResponseCode;
import lombok.extern.slf4j.Slf4j;

@Slf4j
public class CommonException extends RuntimeException {

    private final ResponseCode code;
    private final String customMessage;
    private String redirect = "/";

    public CommonException(ResponseCode code) {
        this.code = code;
        this.customMessage = code.message();
    }

    public CommonException(ResponseCode code, String customMessage) {
        this.code = code;
        this.customMessage = customMessage;
    }

    public CommonException(ResponseCode code, Exception e) {
        super(e);
        this.code = code;
        this.customMessage = code.message();
        log.error(e.getMessage(), e);
    }

    public String redirect(){return redirect; }
    public ResponseCode code() {
        return code;
    }

    @Override
    public String getMessage() {
        return customMessage;
    }
}

File: mail/src/main/kotlin/com/grepp/spring/infra/error/exceptions/CommonException.kt
package com.grepp.spring.infra.error.exceptions

import com.grepp.spring.infra.response.ResponseCode
import org.slf4j.LoggerFactory

open class CommonException : RuntimeException {
    private val code: ResponseCode
    private var redirect = "/"
    private val log = LoggerFactory.getLogger(javaClass)

    constructor(code: ResponseCode) {
        this.code = code
    }

    constructor(code: ResponseCode, redirect: String) {
        this.code = code
        this.redirect = redirect
    }

    constructor(code: ResponseCode, e: Exception) {
        this.code = code
        log.error(e.message, e)
    }

    constructor(code: ResponseCode, e: Exception, redirect: String) {
        this.code = code
        this.redirect = redirect
        log.error(e.message, e)
    }

    fun redirect(): String {
        return redirect
    }

    fun code(): ResponseCode {
        return code
    }
}
